bitcoin/src/address/error.rs

```diff
@@ -124,47 +106,21 @@ impl std::error::Error for ParseError {
         match *self {
             Base58(ref e) => Some(e),
             Bech32(ref e) => Some(e),
-            WitnessVersion(ref e) => Some(e),
-            WitnessProgram(ref e) => Some(e),
-            UnknownHrp(ref e) => Some(e),
-            LegacyAddressTooLong(ref e) => Some(e),
-            InvalidBase58PayloadLength(ref e) => Some(e),
-            InvalidLegacyPrefix(ref e) => Some(e),
             NetworkValidation(ref e) => Some(e),
         }
     }
 }
 
-impl From<base58::Error> for ParseError {
-    fn from(e: base58::Error) -> Self { Self::Base58(e) }
-}
-
-impl From<bech32::segwit::DecodeError> for ParseError {
-    fn from(e: bech32::segwit::DecodeError) -> Self { Self::Bech32(e) }
+impl From<Base58Error> for ParseError {
+    fn from(e: Base58Error) -> Self { Self::Base58(e) }
 }
 
-impl From<witness_version::TryFromError> for ParseError {
-    fn from(e: witness_version::TryFromError) -> Self { Self::WitnessVersion(e) }
-}
-
-impl From<witness_program::Error> for ParseError {
-    fn from(e: witness_program::Error) -> Self { Self::WitnessProgram(e) }
+impl From<Bech32Error> for ParseError {
+    fn from(e: Bech32Error) -> Self { Self::Bech32(e) }
 }
 
 impl From<UnknownHrpError> for ParseError {
-    fn from(e: UnknownHrpError) -> Self { Self::UnknownHrp(e) }
-}
-
-impl From<LegacyAddressTooLongError> for ParseError {
-    fn from(e: LegacyAddressTooLongError) -> Self { Self::LegacyAddressTooLong(e) }
-}
-
-impl From<InvalidBase58PayloadLengthError> for ParseError {
-    fn from(e: InvalidBase58PayloadLengthError) -> Self { Self::InvalidBase58PayloadLength(e) }
-}
-
-impl From<InvalidLegacyPrefixError> for ParseError {
-    fn from(e: InvalidLegacyPrefixError) -> Self { Self::InvalidLegacyPrefix(e) }
+    fn from(err: UnknownHrpError) -> ParseError { ParseError::Bech32(Bech32Error::UnknownHrp(err)) }
```

Comment:
Can be written more simply as:

```rust
 fn from(e: UnknownHrpError) -> ParseError { Self::Bech32(e.into()) }
```

Reply:
Note `e` instead of `err` to fit in with the other code.

Reply:
Done

Reply:
You marked as resolved but this line is still in there, unless I'm mistaken.

Reply:
Whoops, they were still in my editor uncommitted.  Should be there now.